tests/testapp/tests/sync/test_operations.py

```diff
@@ -271,6 +271,320 @@ def test_local_queue_operation__noop(self, mock_queue):
         mock_queue.assert_not_called()
 
 
+@override_settings(MORANGO_SERIALIZE_BEFORE_QUEUING=False, MORANGO_DISABLE_FSIC_V2_FORMAT=False)
+class QueueStoreIntoBufferV2TestCase(TestCase):
+    def setUp(self):
+        super(QueueStoreIntoBufferV2TestCase, self).setUp()
+        self.data = create_dummy_store_data()
+        self.transfer_session = self.data["sc"].current_transfer_session
+        self.context = mock.Mock(
+            spec=LocalSessionContext,
+            transfer_session=self.transfer_session,
+            sync_session=self.transfer_session.sync_session,
+            filter=self.transfer_session.get_filter(),
+            is_push=self.transfer_session.push,
+            is_server=False,
+            capabilities=[FSIC_V2_FORMAT],
+        )
+
+    def test_many_fsics(self):
+        """
+        Testing the limits of how many instances can be included before we hit 'Expression tree is too large'
+        """
+        fsics = {"super": {}, "sub": {"": {self.data["group1_id"].id: 1, self.data["group2_id"].id: 1}}}
+        fsics["sub"][""].update({uuid.uuid4().hex: i for i in range(995)})
+        self.transfer_session.client_fsic = json.dumps(fsics)
+        self.transfer_session.server_fsic = json.dumps({"super": {}, "sub": {}})
+        _queue_into_buffer_v2(self.transfer_session)
+        # ensure all store and buffer records are buffered
+        assertRecordsBuffered(self.data["group1_c1"])
+        assertRecordsBuffered(self.data["group1_c2"])
+        assertRecordsBuffered(self.data["group2_c1"])
+
+    # def test_very_many_fsics(self):
+    #     """
+    #     Regression test against 'Expression tree is too large (maximum depth 1000)' error with many fsics
+    #     """
+    #     fsics = {"super": {}, "sub": {"": {self.data["group1_id"].id: 1, self.data["group2_id"].id: 1}}}
+    #     fsics["sub"][""].update({uuid.uuid4().hex: i for i in range(20000)})
+    #     self.transfer_session.client_fsic = json.dumps(fsics)
+    #     self.transfer_session.server_fsic = json.dumps({"super": {}, "sub": {}})
+    #     _queue_into_buffer_v2(self.transfer_session)
+    #     # ensure all store and buffer records are buffered
+    #     assertRecordsBuffered(self.data["group1_c1"])
+    #     assertRecordsBuffered(self.data["group1_c2"])
+    #     assertRecordsBuffered(self.data["group2_c1"])
```

Comment:
Are these still needed? What were the results of testing how many fsics we can allow?

Reply:
Oh, I thought I had posted this somewhere but I think it seems I probably hadn't: In testing the limits here, it started failing I think around 996 instance_ids. May be sufficient in practice for now, but not a comfortable limit to have longer-term.

Ideally I'd like to re-introduce chunking and uncomment this test before we go from an alpha to a release, but I was thinking in the interim (and perhaps for the S2S pre-release) we could get this alpha integrated into Kolibri for further testing.